src/cargo/sources/git/utils.rs

```diff
@@ -694,12 +695,40 @@ pub fn with_fetch_options(
     let mut progress = Progress::new("Fetch", config);
     network::with_retry(config, || {
         with_authentication(url, git_config, |f| {
+            let mut last_recv = 0.0; // in Byte
+            let mut last_rate = 0.0; // in Byte/s
+            let mut last_update = Instant::now();
             let mut rcb = git2::RemoteCallbacks::new();
             rcb.credentials(f);
-
             rcb.transfer_progress(|stats| {
+                let indexed_deltas = stats.indexed_deltas();
+                let msg = if indexed_deltas > 0 {
+                    // Resolving deltas.
+                    format!(" ({}/{})", indexed_deltas, stats.total_deltas())
```

Comment:
I'm a bit naive as to git's update protocol and such, but is this operation purely a local one? (e.g. one we perform locally after fetching all remote data). I'm thinking it'd be nice that if this involved remote traffic it'd be good if we could still display the transfer rate, but if there's no remote traffic happening here at this point then no worries.

Reply:
AFAIK resolving deltas is just doing some indexing and decompressing against packfile [^1], so there supposed to be no data transfer. I've just tried to disconnect from the internet during the resolving process. It succeeded to clone the registry index as usual.

[^1]: https://stackoverflow.com/a/4690690/8851735

Reply:
Ah ok, perhaps we should include a string of some form with this? Could we match roughly what git does for naming?

Reply:
```
Resolving deltas: 100% (13253/13253), done.
```

That's how git does. We can show the same wording but it will exceed 80 columns easily.  

```
    Updating crates.io index
       Fetch [======>                  ]  40.50%, resolving deltas (1234/282342)
```

Reply:
Looks reasonable to me yeah! We could consider shortening the progress bar here perhaps? Or maybe putting "resolving deltas" at the end so if it's cut off that's the only thing cut off